### PR TITLE
[docs] xpack security improvements from Beats

### PR DESCRIPTION
**Pull request by bmorelli25**
*These are just my notes, sorry for leaving them here... skip down to this https://github.com/elastic/apm-server/pull/2228#issuecomment-499164798 for the end result*
---
Set up index template (no ILM)
* `manage_index_templates`
  * GET /_template/apm* -> 👍

Set up index template (with ILM)
* `manage_index_templates`, `manage_ilm` on cluster
  * Creates only the transaction ilm policy, then errors: `Exiting: failed to check for alias 'apm-7.2.0-transaction': (status=403) : 403 Forbidden:`
* `manage_index_templates`, `manage_ilm` on cluster, `manage` on `apm-*`
  * 👍 

Set up index lifecycle policies:
* `manage_ilm`, `manage` on `apm-*`
  * Fails, looks like we need manage index templates as well
* `manage_index_templates`, `manage_ilm` on cluster, `manage` on `apm-*`
  * Works

Send data from APM Server to ES
With ILM:
To start up and send data, I needed: `manage_ilm`, `manage_index_templates`, `write` on `apm-*`, and `manage` on `apm-*`

Read indices created by APM Server
* apm_user
  * 👍 

View APM UI in Kib
* kibana_user
  * no, "No APM services installed
* `kibana_user` + `read` on `apm-*`
  * 👍

Links:
Built in roles: https://www.elastic.co/guide/en/elastic-stack-overview/current/built-in-roles.html
Built in privileges: https://www.elastic.co/guide/en/elastic-stack-overview/current/security-privileges.html

### CODE REVIEW

**Comment by bmorelli25** [comment]
Finished testing this, you can see my notes above. Here's what I came up with @graphaelli or @simitt can you take a look:

Set up index template (no ILM)
* `manage_index_templates`, `monitor`

Set up index template (with ILM)
* Add `manage_ilm` on cluster, `manage` on `apm-*`

Set up index lifecycle policies:
* `manage_index_templates`, `manage_ilm` on cluster, `manage` on `apm-*`

Read indices created by APM Server
* apm_user

View/use APM UI in Kib
* `kibana_user` + `apm_user`

**Comment by bmorelli25** [comment]
@simitt without `manage_index_templates` I get this in the logs every 5 second, even after templates have been previously set up:
```log
2019-06-07T09:42:52.049-0700    INFO    pipeline/output.go:93   Attempting to reconnect to backoff(elasticsearch(http://localhost:9200)) with 3 reconnect attempt(s)
2019-06-07T09:42:52.049-0700    DEBUG   [elasticsearch] elasticsearch/client.go:711     ES Ping(url=http://localhost:9200)
2019-06-07T09:42:52.051-0700    DEBUG   [elasticsearch] elasticsearch/client.go:734     Ping status code: 200
2019-06-07T09:42:52.052-0700    INFO    elasticsearch/client.go:735     Attempting to connect to Elasticsearch version 7.2.0
2019-06-07T09:42:52.052-0700    DEBUG   [elasticsearch] elasticsearch/client.go:753     GET http://localhost:9200/  <nil>
2019-06-07T09:42:52.054-0700    DEBUG   [elasticsearch] elasticsearch/client.go:753     HEAD http://localhost:9200/_template/apm-7.2.0  <nil>
2019-06-07T09:42:52.055-0700    DEBUG   [template]      template/load.go:214    Load default fields
2019-06-07T09:42:52.083-0700    INFO    template/load.go:108    Try loading template apm-7.2.0 to Elasticsearch
2019-06-07T09:42:52.084-0700    DEBUG   [elasticsearch] elasticsearch/client.go:753     PUT http://localhost:9200/_template/apm-7.2.0 
```

Trying to send data produces a `request failed: request failed with 503 Service Unavailable: {"accepted":0,"errors":[{"message":"queue is full"}]}`

File: docs/copied-from-beats/security/basic-auth.asciidoc [hunk 1]
```diff
@@ -6,168 +6,55 @@ When sending data to a secured cluster through the `elasticsearch`
...
---------------------------------------------------
-<1> You created this user earlier.
+----
+<1> Let's assume this user has the privileges required to publish events to
+{es}.
 <2> The example shows a hard-coded password, but you should store sensitive
 values in the <<keystore,secrets keystore>>.
+--
++
+If you've configured the {kib} endpoint, also specify credentials for
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/copied-from-beats/security/basic-auth.asciidoc]: Good catch

**Comment by bmorelli25** [comment]
Removed Kibana info and added ingest pipelines: `monitor` on cluster and `ingest_admin` role

**Comment by simitt** [comment]
@bmorelli25 I run some tests regarding the permissions. Ingesting data with an apm-server user that does not have the `manage_index_template` ability is possible when the setup is already run by another user with more privileges and `setup.template.enabled` is set to `false` during ingestion time. 

Creating roles and users is fixed now in the integration testing, so here's how I tested:
fired up integration testing instances with `--xpack-secure` and run a `apm-server setup` against it with `apm-server.ilm.enabled=true`. Create a new role and a new user that does not have the `manage_index_templates` privilege, and configure ES output for the apm-server with this user. Set `setup.template.enabled=false`, run `apm-server -e` and ingest some data.

I also found this documentation in [filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/feature-roles.html#privileges-to-publish-events). WDYT about documenting this somehow similar?

**Comment by bmorelli25** [comment]
@simitt, is totally right. I didn't try setting `setup.template.enabled=false`. Changing that setting allows me to ingest data without `manage_index_template`. Will make the update!

**Pull request merged**